lib/node/http.js

```diff
@@ -297,13 +297,30 @@ class HTTP extends Server {
     this.get('/mempool', async (req, res) => {
       enforce(this.mempool, 'No mempool available.');
 
-      const hashes = this.mempool.getSnapshot();
-      const result = [];
+      const valid = Validator.fromRequest(req);
+      const details = valid.bool('details', false);
 
-      for (const hash of hashes)
-        result.push(util.revHex(hash));
+      if (details) {
+        const txs = this.mempool.getHistory();
 
-      res.json(200, result);
+        // stream the response
+        res.setType('json');
+        res.write('[' + JSON.stringify(txs[0].toJSON()));
```

Comment:
`getJSON(this.network)` returns the right network address.

Reply:
Good catch